portal-kernel/src/com/liferay/portal/kernel/service/persistence/impl/BasePersistenceImpl.java

```diff
@@ -133,6 +156,86 @@ public long countWithDynamicQuery(
 		return firstResult.longValue();
 	}
 
+	@Override
+	@SuppressWarnings("unchecked")
+	public <R> R dslQuery(DSLQuery dslQuery) {
+		Session session = null;
+
+		try {
+			session = openSession();
+
+			DefaultASTNodeListener defaultASTNodeListener =
+				new DefaultASTNodeListener();
+
+			String[] tableNames = defaultASTNodeListener.getTableNames();
```

Comment:
@Preston-Crary this can not be right, the defaultASTNodeListener is still empty at this moment, there won't be any table names in it.
Please make sure the test covers this too.

Reply:
Yeah, should have updated the something to cover the new method.  I'll update change-tracking-service to use the new API.